source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -264,8 +264,16 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
     idle_timeout_ = absl::nullopt;
   }
 
-  // If we are provided a different request_id_extension implementation to use try and create a new
-  // instance of it, otherwise use default one.
+  if (config.strip_any_host_port()) {
+    strip_port_type_ = Http::StripPortType::Any;
+  } else if (config.strip_matching_host_port()) {
```

Comment:
Given that defaults are false, can we check if both are true and throw an error? I think this is safe to do and will approximate the future oneof?

Reply:
Yes.  We can do that.  Changed and added a test. PTAL